blocks/library/embed/index.js

```diff
@@ -22,6 +26,23 @@ registerBlock( 'core/embed', {
 	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, title, caption } = attributes;
 
+		if ( ! url ) {
```

Comment:
I don't know that this necessarily needs to be built-in, but I think we could surface better the fact that `edit` is truly a React component, and can itself branch into separate subcomponents up to the implementer's discretion.